src/responder/common/data_provider/rdp_message.c

```diff
@@ -269,6 +269,10 @@ static void rdp_message_send_and_reply_done(DBusPendingCall *pending,
     sbus_req = talloc_get_type(ptr, struct sbus_request);
 
     ret = rdp_process_pending_call(sbus_req, pending, &reply);
+    if (ret != EOK) {
+        /* Something bad happened. Just kill the request. */
+        goto done;
+    }
     if (reply == NULL) {
```

Comment:
I am little confused. It is more about philosophy of code.

I discussed similar issue with some members of sssd team. The question is what is more important if error code or returned values. I show short example:

Let we have function call like:
```
errno_t ret = my_function(input_argument, &output_data); 
```
Is better to rely on value of ret or on value of output_data?

Our result was if ```ret != EOK``` it we should assume that ```output_data``` could be corrupted.

I understand that this comment is not strictly connected to recent issue. But, I would like to ask: "Do we have genral rule for those situations?"